### PR TITLE
Track Metrics Series of functions

### PR DESCRIPTION
**Pull request by marosteg**
track.avgdist(), track.cumdist(), and track.stepdist() provide the average distance travelled per time step, cumulative distance travelled over the track, and a vector of the distance travelled in each time step, respectively (this final one is intended to enable extraction of any additional desired metrics)

### CODE REVIEW

**Comment by camrinbraun** [comment]
@marosteg thanks for pushing these speeds functions. two questions:
1) are they well tested and seem to be working well? ready for merge?

2) can you copy/paste a bit of your run script code here that shows how they're implemented? don't need a MWE, just a code block that shows how you're using them for our future reference.

File: R/km.per.gridunit.R [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#' Input Speed Parameter Guidance by Martin C. Arostegui aka Dr. Mola
+#' km.per.gridunit() calculates the min and max longitudinal width (in km) of a single grid unit
```
Comment (camrinbraun) [inline R/km.per.gridunit.R:2]: there's actually an @author tag, for example see https://github.com/camrinbraun/HMMoce/blob/dev/R/gausskern.R

**Comment by marosteg** [comment]
Example:
Input Speed Parameter Guidance (Before Running the Model)
```
> sp.lim <- list(lonmin = -10, lonmax = 37, latmin = 30, latmax = 46)
> km.per.gridunit(limits=sp.lim, res=0.08)
[1] 6.20 7.72
```
So if the literature maximum displacement for a species was 100 km/d, you would take the result (either of the range ends  [provided: 6.20 | 7.72] or the midpoint [you do simple avg: 6.96] and divide 100 by that number to know how many grid units your animal would need to cross to reach ~100 km in a 24 hr time step model. If you did finer time steps, then you need to divide the literature max of km/d to the appropriate fraction at the start and proceed as above.

If using the average: 100 / 6.96 = 14.36782. I then would provide that number of grid units, rounded to a whole number (14), as the upper bound of the estimation window for the migratory speed parameter). These speed inputs to HMMoce are APPROXIMATE given that the km travelled per grid unit longitudinally changes with latitude.
_____________________________

Track Series Metrics (After Running the Model)
Gives output metrics about your track in kilometers (km)
```
tr <- calc.track(s, g = L.res$g, dateVec, iniloc, method='mean')
track.avgdist(track = tr) # calculates the average distance travelled per model time step of a track
track.cumdist(track = tr) # calculates the cumulative distance travelled over a tag deployment track
track.stepdist(track = tr) # calculates a vector of the distance travelled per time step of a tag deployment track
```
These are all tested and ready to roll.

**Comment by camrinbraun** [comment]
note that this got merged to master not dev so changes are reflected in master only. we need to merge dev and master soon anyway...

**Pull request merged**